crates/tx/src/action.rs

```diff
@@ -63,11 +63,13 @@ pub enum PgfAction {
     UpdateStewardCommission(Address),
 }
 
-/// MASP tx action.
+/// MASP tx actions.
 #[derive(Clone, Debug, BorshDeserialize, BorshSerialize)]
-pub struct MaspAction {
+pub enum MaspAction {
     /// The hash of the masp [`crate::types::Section`]
-    pub masp_section_ref: Hash,
+    MaspSectionRef(Hash),
+    /// A required signer for the transaction
+    MaspSigner(Address),
```

Comment:
do we need this new case? It looks like it's never constructed

Reply:
Ah yes sorry. This is not used at the moment cause it's only needed when a change in the balance of an address does not match what we see in the `TranpsarentBundle` of the `Masp` section. This in turn can only happen if a transaction is both a masp and a transparent transfer which in up until this PR is something that we don't support in the SDK and client. But it should become relevant after #3459